tsl/test/shared/expected/dist_remote_error.out

```diff
@@ -111,15 +110,42 @@ select table_name from create_distributed_hypertable('metrics_dist_br',
  metrics_dist_br
 (1 row)
 
--- Test that the insert fails on data nodes.
+-- Test that INSERT and COPY fail on data nodes.
+-- Note that we use the text format for the COPY input, so that the access node
+-- doesn't call `recv` and fail by itself. It's going to use binary format for
+-- transfer to data nodes regardless of the input format.
+-- First, create the reference.
+\copy (select * from metrics_dist_remote_error) to 'dist_remote_error.text' with (format text);
+-- We have to test various interleavings of COPY and INSERT to check that
+-- one can recover from connection failure states introduced by another.
+\copy metrics_dist_br from 'dist_remote_error.text' with (format text);
+ERROR:  [data_node_2]: debug point: requested to error out after 7103 rows, 7103 rows seen
+\copy metrics_dist_br from 'dist_remote_error.text' with (format text);
+ERROR:  [data_node_2]: debug point: requested to error out after 7103 rows, 7103 rows seen
+insert into metrics_dist_br select * from metrics_dist_remote_error;
+ERROR:  [data_node_2]: debug point: requested to error out after 7103 rows, 7103 rows seen
 insert into metrics_dist_br select * from metrics_dist_remote_error;
 ERROR:  [data_node_2]: debug point: requested to error out after 7103 rows, 7103 rows seen
--- Also test that the COPY fails on data nodes. Note that we use the text format
--- for the insert, so that the access node doesn't call `recv` and fail by itself.
--- It's going to use binary format for transferring COPY data to data nodes
--- regardless of the input format.
-\copy (select * from metrics_dist_remote_error) to 'dist_remote_error.text' with (format text);
 \copy metrics_dist_br from 'dist_remote_error.text' with (format text);
 ERROR:  [data_node_2]: debug point: requested to error out after 7103 rows, 7103 rows seen
 drop table metrics_dist_br;
+-- Table with sleepy receive for a data type, to improve coverage of the waiting
+-- code on the access node.
+create table metrics_dist_bl(like metrics_dist);
+alter table metrics_dist_bl alter column v0 type bl;
+select table_name from create_distributed_hypertable('metrics_dist_bl',
+    'time', 'device_id');
+   table_name    
+ metrics_dist_bl
+(1 row)
+
+-- Test INSERT and COPY with slow data node.
+\copy metrics_dist_bl from 'dist_remote_error.text' with (format text);
+insert into metrics_dist_bl select * from metrics_dist_remote_error;
+select count(*) from metrics_dist_bl;
+ count 
+ 40000
```

Comment:
That would be 40 second test to run, right? I hope that's not too long

Reply:
The entire test runs about 20 s, because only about 5k rows go to the third data node where the recv() is sleepy.